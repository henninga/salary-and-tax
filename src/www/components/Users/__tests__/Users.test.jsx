import mockComponent from 'identity-obj-proxy';
import React from 'react';
import renderer from 'react-test-renderer';
import { UsersComponent } from '../index';


const userList = [
    {
        name: 'CV',
        userId: 'partnerCv',
        cvId: 'idCv',
    },
    {
        name: 'Partner',
        userId: 'cvPartner',
        cvId: 'cvId',
    },
];

jest.mock('../../Ui', () => mockComponent);

const mockGetUsers = jest.fn();
const mockSelectUser = jest.fn();

describe('UsersComponent', () => {
    it('should render', () => {
        expect(renderer.create(<UsersComponent
          list={userList}
          getUsers={mockGetUsers}
          isFetching={false}
          isLoaded
          selectUser={mockSelectUser}
          selectedUser={{}}
        />).toJSON()).toMatchSnapshot();
    });
    it('should render loading', () => {
        expect(renderer.create(<UsersComponent
          list={[]}
          getUsers={mockGetUsers}
          isFetching
          isLoaded={false}
          selectUser={mockSelectUser}
          selectedUser={{}}
        />).toJSON()).toMatchSnapshot();
    });
    it('should render empty', () => {
        expect(renderer.create(<UsersComponent
          list={[]}
          getUsers={mockGetUsers}
          isFetching={false}
          isLoaded
          selectUser={mockSelectUser}
          selectedUser={{}}
        />).toJSON()).toMatchSnapshot();
    });
    it('should dispatch when mounted', () => {
        renderer.create(<UsersComponent
          list={userList}
          getUsers={mockGetUsers}
          isFetching={false}
          isLoaded
          selectUser={mockSelectUser}
          selectedUser={{}}
        />);
        expect(mockGetUsers).toHaveBeenCalled();
    });
});

export default userList;
