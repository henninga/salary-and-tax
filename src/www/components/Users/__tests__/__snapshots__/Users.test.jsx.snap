// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsersComponent should render 1`] = `
<div>
  <Dropdown
    filter={true}
    onChange={[MockFunction]}
    optionLabel="name"
    options={
      Array [
        Object {
          "cvId": "idCv",
          "name": "CV",
          "userId": "partnerCv",
        },
        Object {
          "cvId": "cvId",
          "name": "Partner",
          "userId": "cvPartner",
        },
      ]
    }
    style={
      Object {
        "width": "100%",
      }
    }
    value={Object {}}
  />
</div>
`;

exports[`UsersComponent should render empty 1`] = `
<div>
  <h2>
    No users found.
  </h2>
</div>
`;

exports[`UsersComponent should render loading 1`] = `
<div>
  <h2>
    Loading...
  </h2>
</div>
`;
